backend/src/main/java/com/daedan/festabook/notification/infrastructure/FCMNotificationManager.java

```diff
@@ -0,0 +1,81 @@
+package com.daedan.festabook.notification.infrastructure;
+
+import com.daedan.festabook.global.exception.BusinessException;
+import com.daedan.festabook.notification.dto.NotificationMessage;
+import com.daedan.festabook.organization.domain.OrganizationNotificationManager;
+import com.daedan.festabook.place.domain.PlaceNotificationManager;
+import com.google.firebase.messaging.FirebaseMessaging;
+import com.google.firebase.messaging.FirebaseMessagingException;
+import com.google.firebase.messaging.Message;
+import com.google.firebase.messaging.Notification;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FCMNotificationManager implements OrganizationNotificationManager, PlaceNotificationManager {
+
+    private static final String ORGANIZATION_TOPIC_PREFIX = "notifications-organization-";
+    private static final String PLACE_TOPIC_PREFIX = "notifications-place-";
+
+    @Override
+    public void subscribeOrganizationTopic(Long organizationId, String fcmToken) {
+        String topic = ORGANIZATION_TOPIC_PREFIX + organizationId;
+        subscribeTopic(topic, fcmToken);
+    }
+
+    @Override
+    public void unsubscribeOrganizationTopic(Long organizationId, String fcmToken) {
+        String topic = ORGANIZATION_TOPIC_PREFIX + organizationId;
+        unsubscribeTopic(topic, fcmToken);
+    }
+
+    @Override
+    public void sendToOrganizationTopic(Long organizationId, NotificationMessage notificationMessage) {
+        String topic = ORGANIZATION_TOPIC_PREFIX + organizationId;
+        sendToTopic(topic, notificationMessage);
+    }
+
+    @Override
+    public void subscribePlaceTopic(Long placeId, String fcmToken) {
+        String topic = PLACE_TOPIC_PREFIX + placeId;
+        subscribeTopic(topic, fcmToken);
+    }
+
+    @Override
+    public void unsubscribePlaceTopic(Long placeId, String fcmToken) {
+        String topic = PLACE_TOPIC_PREFIX + placeId;
+        unsubscribeTopic(topic, fcmToken);
+    }
+
+    protected void subscribeTopic(String topic, String fcmToken) {
+        try {
+            FirebaseMessaging.getInstance().subscribeToTopic(List.of(fcmToken), topic);
```

Comment:
getInstance()는 상수로 추출하는게 어떤가요?

```suggestion
private static final FIREBASE_MESSAGING = FirebaseMessaging.getInstance();
```

Reply:
의도는 이해했는데, 저희가 현재 구조상 FcmNotificationManager가 엄청나게 길어질 걸 대비해서 위 아래로 확인하는 비용이 좀 생길 것 같긴한데... 어떻게 생각하시나요?

Reply:
@eoehd1ek 해당 부분은 후유가 판단하고 적용해주시면 될 것 같아요!